app/governance/proposal/page.tsx

```diff
@@ -0,0 +1,315 @@
+"use client";
+
+import { useRouter, useSearchParams } from "next/navigation";
+import Text from "@/components/text";
+import styles from "./proposalModal.module.scss";
+import {
+  calculateVotePercentages,
+  formatDeposit,
+  formatProposalStatus,
+  formatProposalType,
+  formatTime,
+} from "@/utils/gov/formatData";
+import Icon from "@/components/icon/icon";
+import Button from "@/components/button/button";
+import useProposals from "@/hooks/gov/useProposals";
+import { useState } from "react";
+import { Proposal } from "@/hooks/gov/interfaces/proposal";
+import useCantoSigner from "@/hooks/helpers/useCantoSigner";
+import Splash from "@/components/splash/splash";
+import { VoteOption } from "@/hooks/gov/interfaces/voteOptions";
+import Image from "next/image";
+import { NEW_ERROR } from "@/config/interfaces";
+
+import { VotingInfoBox } from "../components/VotingInfoBox/VotingInfoBox";
+import { TransactionFlowType } from "@/transactions/flows/flowMap";
+import { NewTransactionFlow } from "@/transactions/flows/types";
+import Spacer from "@/components/layout/spacer";
+
+export default function Page() {
+  function castVote(proposalId: number, voteOption: VoteOption | null) {
+    if (signer) {
+      if (!voteOption) {
+        return NEW_ERROR("Please select a vote option");
+      }
+      //console.log("cast vote test");
+      const newFlow: NewTransactionFlow = {
```

Comment:
we should define this function in the transactions folder and just call it here instead.